dashboard/public/pixelation/pixelation.js

```diff
@@ -48,78 +60,92 @@ function customizeStyles() {
     }
 
     // The layout is fundamentally different in version 1 than it is in other versions.
     // Rearrange the DOM so that the visualization column sits at the top left.
-    var visualizationColumn = document.getElementById('visualizationColumn');
-    var visualizationEditorHeader = document.getElementById('visualizationEditorHeader');
-    visualizationColumn.parentNode.insertBefore(visualizationColumn, visualizationEditorHeader);
-  } else if (options.version === '2') {
-    $('.hide_on_v2').hide();
-    $('#height, #width').prop('readonly', true);
+    var visualizationColumn = document.getElementById("visualizationColumn");
+    var visualizationEditorHeader = document.getElementById(
+      "visualizationEditorHeader"
+    );
+    visualizationColumn.parentNode.insertBefore(
+      visualizationColumn,
+      visualizationEditorHeader
+    );
+  } else if (options.version === "2") {
+    $(".hide_on_v2").hide();
+    $("#height, #width").prop("readonly", true);
   }
   if (isHexLevel()) {
-    $('input[name="binHex"][value="hex"]').prop('checked', true);
+    $('input[name="binHex"][value="hex"]').prop("checked", true);
   }
-  if (options.hideEncodingControls === true || options.hideEncodingControls === 'true') {
-    $('.encoding_controls').hide();
+  if (
+    options.hideEncodingControls === true ||
+    options.hideEncodingControls === "true"
+  ) {
+    $(".encoding_controls").hide();
   }
   if (options.shortInstructions) {
-    $('#below_viz_instructions').text(options.shortInstructions).show();
+    $("#below_viz_instructions")
+      .text(options.shortInstructions)
+      .show();
   }
 }
 
 function initProjects() {
   // Initialize projects for save/load functionality if channel id is present.
   if (appOptions.channel) {
     if (!window.dashboard) {
-      throw new Error('Assume existence of window.dashboard');
+      throw new Error("Assume existence of window.dashboard");
     }
 
     var sourceHandler = {
-      setMakerAPIsEnabled: function (_) {},
-      getMakerAPIsEnabled: function () {
+      setMakerAPIsEnabled: function(_) {},
+      getMakerAPIsEnabled: function() {
         return false;
       },
-      setSelectedSong: function () {},
-      getSelectedSong: function () {
+      setSelectedSong: function() {},
+      getSelectedSong: function() {
         return false;
       },
-      setInitialLevelHtml: function (levelHtml) {},
-      getLevelHtml: function () {
-        return '';
+      setInitialLevelHtml: function(levelHtml) {},
+      getLevelHtml: function() {
+        return "";
       },
-      setInitialAnimationList: function () {},
-      getAnimationList: function (callback) {
+      setInitialAnimationList: function() {},
+      getAnimationList: function(callback) {
         callback({});
       },
-      setInitialLevelSource: function (levelSource) {
+      setInitialGeneratedProperties: function(_) {},
+      getGeneratedProperties() {
+        return undefined;
+      },
```

Comment:
This broke on IE11; because this is non-transpiled code, and IE11 is the only browser that doesn't natively support this syntax yet.  ([Slack thread](https://codedotorg.slack.com/archives/C0T0PNTM3/p1562710353415000))

Reply:
Fix-forward in commit https://github.com/code-dot-org/code-dot-org/commit/173e7d39f190e69f93f350e053c4b3ea10c8678c.